# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
# Commit: 3ca24e761cc3
# Full Hash: 3ca24e761cc31bbb71c659661874a995ef3d3db1
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Remove WebrtcGmpPCHandleSetter and plumb pc handles explicitly. r=bwc
#   
#   With the libwebrtc update codecs will be created off-main.
#   WebrtcGmpPCHandleSetter not only adds a lot of indirection, it is also
#   main-thread-only. With this patch we can create gmp codecs on any thread.
# ==============================================================================

diff -r 1913c332695e -r 3ca24e761cc3 dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
--- a/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h	Wed Feb 10 09:58:11 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h	Wed Feb 10 10:34:14 2021 +0100
@@ -13,6 +13,9 @@
 namespace mozilla {
 class WebrtcVideoDecoderFactory : public webrtc::VideoDecoderFactory {
  public:
+  explicit WebrtcVideoDecoderFactory(std::string aPCHandle)
+      : mPCHandle(std::move(aPCHandle)) {}
+
   std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
     MOZ_CRASH("Unexpected call");
     return std::vector<webrtc::SdpVideoFormat>();
@@ -20,11 +23,17 @@
 
   std::unique_ptr<webrtc::VideoDecoder> CreateVideoDecoder(
       const webrtc::SdpVideoFormat& aFormat) override;
+
+ private:
+  const std::string mPCHandle;
 };
 
 class WebrtcVideoEncoderFactory : public webrtc::VideoEncoderFactory {
   class InternalFactory : public webrtc::VideoEncoderFactory {
    public:
+    explicit InternalFactory(std::string aPCHandle)
+        : mPCHandle(std::move(aPCHandle)) {}
+
     std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
       MOZ_CRASH("Unexpected call");
       return std::vector<webrtc::SdpVideoFormat>();
@@ -34,11 +43,14 @@
         const webrtc::SdpVideoFormat& aFormat) override;
 
     bool Supports(const webrtc::SdpVideoFormat& aFormat);
+
+   private:
+    const std::string mPCHandle;
   };
 
  public:
-  WebrtcVideoEncoderFactory()
-      : mInternalFactory(MakeUnique<InternalFactory>()) {}
+  explicit WebrtcVideoEncoderFactory(std::string aPCHandle)
+      : mInternalFactory(MakeUnique<InternalFactory>(std::move(aPCHandle))) {}
 
   std::vector<webrtc::SdpVideoFormat> GetSupportedFormats() const override {
     MOZ_CRASH("Unexpected call");
